### PR TITLE
feat: Make FullEvent serializeable

### PR DESCRIPTION
**Pull request by cheesycod**

### CODE REVIEW

**Comment by jamesbt365** [comment]
...Whats the usecase here? Seems like you're trying to do something the library isn't designed for and you would probably have a better experience using twilight instead of creating a hacky solution with serenity.

**Comment by GnomedDev** [comment]
I've talked in DMs about this, it's for serializing to an interpreter and removes 3k LOC with the only downside being slightly longer fresh compile times. I'm good with this.

**Comment by GnomedDev** [comment]
I'll enable auto merge, but you can have the final say on approving. @jamesbt365

File: src/model/id.rs [hunk 1]
```diff
@@ -339,7 +339,7 @@ id_u64! {
 /// This identifier is special, it simply models internal IDs for type safety,
 /// and therefore cannot be [`Serialize`]d or [`Deserialize`]d.
```
Comment (jamesbt365) [inline src/model/id.rs]: If you're gonna do it, this comment is outdated?

Comment (GnomedDev) [reply src/model/id.rs]: Yep, this comment should be `cannot be deserialized.`

Comment (cheesycod) [reply src/model/id.rs]: Fixed

**Review by jamesbt365 — APPROVED** [review]

**Pull request merged**